drivers/hmis/app/graphql/mutations/ce/calculate_client_eligibility.rb

```diff
@@ -0,0 +1,83 @@
+###
+# Copyright 2016 - 2025 Green River Data Analysis, LLC
+#
+# License detail: https://github.com/greenriver/hmis-warehouse/blob/production/LICENSE.md
+###
+
+# frozen_string_literal: true
+
+module Mutations
+  class Ce::CalculateClientEligibility < CleanBaseMutation
+    description 'Calculate client eligibility based on provided assessment values and return applicable project types'
+
+    argument :enrollment_id, ID, required: true
+    argument :form_definition_identifier, String, required: true
+    argument :values_by_link_id, Types::JsonObject, required: true
+
+    field :project_types, [Types::HmisSchema::Enums::ProjectType], null: false
+
+    def resolve(enrollment_id:, form_definition_identifier:, values_by_link_id:)
+      enrollment = Hmis::Hud::Enrollment.viewable_by(current_user).find(enrollment_id)
+      access_denied! unless current_user.can_edit_enrollments_for?(enrollment)
+
+      client = enrollment.client.destination_client
+      return { project_types: [] } unless client
```

Comment:
From a UX perspective, this button should ideally work regardless of whether the client has a destination client yet -- after all, it's basing eligibility off of the current form values. But it's more complicated than that internally, because our match engine expects a destination client, and the current form values are just overrides. I'd like to discuss possible solutions (and whether they're worthwhile to pursue)